packages/server/lib/socket-base.ts

```diff
@@ -401,6 +403,16 @@ export class SocketBase {
           return socket.emit('dev-server:on-spec-updated')
         })
 
+        socket.on('studio:init', async (cb) => {
+          await options.onStudioInit()
```

Comment:
Can these calls fail? Do we need to ensure the callbck is called (maybe with the error)?

Reply:
I think everything I'm calling is fairly protected, but wouldn't hurt to wrap it with a try catch and pass along the error  if needed.

Reply:
Updated: [`e619cb2` (#31239)](https://github.com/cypress-io/cypress/pull/31239/commits/e619cb23c7d692f22ef0e29370b9b5d8555e6a72)